src/main/java/com/github/fishio/audio/FishClip.java

```diff
@@ -0,0 +1,113 @@
+package com.github.fishio.audio;
+
+import java.io.IOException;
+
+import javax.sound.sampled.Clip;
+import javax.sound.sampled.FloatControl;
+import javax.sound.sampled.LineEvent;
+import javax.sound.sampled.LineListener;
+import javax.sound.sampled.LineEvent.Type;
+
+/**
+ * A decorator for {@link Clip}, which adds functionality to check if the
+ * clip is done playing.
+ */
+public class FishClip implements LineListener, AutoCloseable {
+	private Clip clip;
+	private volatile boolean done;
+	private boolean effect;
+	
+	/**
+	 * @param clip
+	 * 		the clip this FishClip should represent.
+	 * @param effect
+	 * 		if this clip is a sound effect (true) or music (false).
+	 */
+	public FishClip(Clip clip, boolean effect) {
+		this.clip = clip;
+		this.effect = effect;
+		clip.addLineListener(this);
+		AudioEngine.getInstance().registerVolumeListener(effect, () -> updateVolume());
+		
+		//Set the initial volume
+		updateVolume();
+	}
+	
+	/**
+	 * @return
+	 * 		the clip this FishClip represents.
+	 */
+	public Clip getClip() {
+		return clip;
+	}
+	
+	/**
+	 * @return
+	 * 		<code>true</code> if this clip is a sound effect.<br>
+	 * 		<code>false</code> if it is (background) music.
+	 */
+	public boolean isEffect() {
+		return effect;
+	}
+	
```

Comment:
See my comment at the isEffect() method in the Sound class.

<organisation,review